src/test/scala/com/microsoft/hyperspace/index/rules/FilterIndexRuleTest.scala

```diff
@@ -163,18 +176,51 @@ class FilterIndexRuleTest extends HyperspaceSuite {
               _,
               _,
               _))) =>
-        val allIndexes = IndexCollectionManager(spark).getIndexes(Seq(Constants.States.ACTIVE))
-        val expectedLocation = getIndexDataFilesPath(indexName)
-        assert(newLocation.rootPaths.head.equals(expectedLocation), "Invalid location.")
-        assert(newPartitionSchema.equals(new StructType()), "Invalid partition schema.")
-        assert(dataSchema.equals(allIndexes.head.schema), "Invalid schema.")
-        assert(bucketSpec.isEmpty, "Invalid bucket spec.")
+        verifyIndexProperties(indexName1, newLocation, newPartitionSchema, dataSchema, bucketSpec)
+        assert(dataSchema.fieldNames.toSet.equals(l.output.map(_.name).toSet))
+
+      case _ => fail("Unexpected plan.")
+    }
+  }
+
+  private def verifyTransformedPlanWithIndex2(logicalPlan: LogicalPlan): Unit = {
```

Comment:
Can we remove some duplication in `verify..Index1` and `verify..Index2`? I was thinking the following:

```
def verifyIndex(logicalPlan, indexName): Unit = {
  val logicalRelation = logicalPlan.collect {
      case l : LogicalRelation => l
   }.head
   logicalRelation match {
      case l@LogicalRelation(HadoopRelation(newLocation..........ParquetFileFormat.....) => 
          assert(condition1)
          assert(condition2)
          assert(index name related checks)
          ...
       case _ => fail("Unexpected plan")
   }
}
```
basically parameterize indexname in the function.
The only issue is we don't explicitly check if the plan is a Project or a Filter. I guess it's ok. 
Let me know if this makes sense or if there's any suggestion to deduplicate. If we want to keep Project and Filter check, it's ok too.

Reply:
Done.